plugins/source/gcp/client/client.go

```diff
@@ -142,16 +143,18 @@ func logProjectIds(logger *zerolog.Logger, projectIds []string) {
 }
 
 // getProjectsV1 requires the `resourcemanager.projects.get` permission to list projects
-func getProjectsV1(ctx context.Context, options ...option.ClientOption) ([]string, error) {
+func getProjectsV1(ctx context.Context, filter string, options ...option.ClientOption) ([]string, error) {
 	var (
 		projects []string
 	)
 	service, err := crmv1.NewService(ctx, options...)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create cloudresourcemanager service: %w", err)
 	}
-
-	call := service.Projects.List().Filter("lifecycleState=ACTIVE").Context(ctx)
+	if filter == "" {
```

Comment:
Can we move the default logic to one place (I know not related to this PR specifcally). but let's do a 

func (*Spec) SetDefaults()